### PR TITLE
Enabling the creation of JavaScript SDK for the Communication resource manager

### PR DESCRIPTION
**Pull request by memontic-ms**
Enabling the creation of JavaScript SDK for the Communication resource manager.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by usvoyager — DISMISSED** [review]
:shipit:

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/communication/resource-manager/readme.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ swagger-to-sdk:
   - repo: azure-sdk-for-python
   - repo: azure-sdk-for-python-track2
   - repo: azure-sdk-for-go
+  - repo: azure-sdk-for-js
     after_scripts:
       - bundle install && rake arm:regen_all_profiles['azure_mgmt_vsonline']
```
Comment (qiaozha) [inline specification/communication/resource-manager/readme.md:58]: JS SDK doesn't have after scripts, you should remove the after scripts or put it right after the repo that needs it.

File: specification/communication/resource-manager/readme.typescript.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## TypeScript
+
+These settings apply only when `--typescript` is specified on the command line.
+Please also specify `--typescript-sdks-folder=<path to the root directory of your azure-sdk-for-js clone>`.
+
+``` yaml $(typescript)
+typescript:
+  azure-arm: true
+  package-name: "@azure/arm-communication"
+  output-folder: "$(typescript-sdks-folder)/sdk/communication/arm-communication"
```
Comment (qiaozha) [inline specification/communication/resource-manager/readme.typescript.md:10]: I just found that the folder https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/communication is not empty, Are those the communication data-plane SDKs?

**Review by qiaozha — COMMENTED** [review]
just have two questions. see my comments

File: specification/communication/resource-manager/readme.typescript.md (hunk 2 above)
Comment (memontic-ms) [reply specification/communication/resource-manager/readme.typescript.md:10]: Correct. Those are the data-plane SDKs.

File: specification/communication/resource-manager/readme.md (hunk 1 above)
Comment (memontic-ms) [reply specification/communication/resource-manager/readme.md:58]: Good catch. That after_script is not needed. I'll remove it.

**Review by qiaozha — APPROVED** [review]

**Pull request merged**